src/CachePlugin.php

```diff
@@ -160,6 +163,22 @@ private function getMaxAge(ResponseInterface $response)
             return (new \DateTime($header))->getTimestamp() - (new \DateTime())->getTimestamp();
         }
 
-        return $this->defaultTtl;
+        return $this->config['default_ttl'];
+    }
+
+    /**
+     * @param array $config
+     *
+     * @return array
+     */
+    protected function configure(array $config = [])
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults($this->config);
```

Comment:
i would prefer to have a `private function getOptionsResolver` that prepares the resolver with our defaults and the constructor assigns the resolve() result to $this->config.

not sure whether there is some sort of best practice around this - there are a couple of ways to do it. could also be all in the constructor.